File: src/BibliotecaCorporativa/backend/BibCorp.Application/Dtos/Emprestimos/Emprestimo.cs
using BibCorp.Application.Dto.Acervos;
using BibCorp.Application.Dto.Patrimonios;

namespace BibCorp.Application.Dto.Emprestimos
{
  public class EmprestimoDto
  {
    public int Id { get; set; }
    public string UserName { get; set; }
    public StatusDto Status { get; set; }
    public string DataEmprestimo { get; set; }
    public string DataPrevistaDevolucao { get; set; }
    public int QtdeDiasEmprestimo { get; set; }
    public string DataDevolucao { get; set; }
    public int QtdeDiasAtraso { get; set; }
    public int AcervoId { get; set; }
    public AcervoDto Acervo { get; set; }
    public int PatrimonioId { get; set; }
    public PatrimonioDto Patrimonio { get; set; }
    public string LocalDeColeta { get; set; }
    public string LocalDeEntrega { get; set; }
  }

  public enum StatusDto
  {
    Reservado = 1,
    Emprestado = 2,
    Devolvido = 3,
    Renovado = 4
  }
}

File: src/BibliotecaCorporativa/frontend/BibCorp-app/src/app/models/Emprestimos/Emprestimo.ts
import { type Patrimonio } from '../../patrimonios/models/patrimonio/Patrimonio'
import { type Acervo } from '../Acervos/Acervo'

export class Emprestimo {
  id: number
  userName: string
  status: Status
  dataEmprestimo: string
  dataPrevistaDevolucao: string
  qtdeDiasEmprestimo: number
  dataDevolucao: string
  qtdeDiasAtraso: number
  acervoId: number
  acervo: Acervo
  patrimonioId: number
  patrimonio: Patrimonio
  localDeColeta: string
  localDeEntrega: string
}

export enum Status{
  Reservado = 1,
  Emprestado = 2,
  Devolvido = 3,
  Renovado = 4,
  Recusado = 5
}
